nomad/structs/structs.go

```diff
@@ -7377,6 +7416,37 @@ func (a *Allocation) ComparableResources() *ComparableResources {
 	}
 }
 
+// COMPAT(0.11): Remove in 0.11
+// CompatibleNetworkResources returns network resources on the allocation
+// by reading AllocatedResources which are populated starting in 0.9 and
+// falling back to pre 0.9 fields (Resources/TaskResources) if set
+func (a *Allocation) CompatibleNetworkResources() []*NetworkResource {
+	var ret []*NetworkResource
```

Comment:
because ComparableResources ignores networks entirely. I didn't know how safe it was to change that given how much its used everywhere so made something very specific for preemption.